src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -2,25 +2,28 @@
 
 import org.skypro.skyshop.product.Product;
 
-import java.util.Iterator;
-import java.util.LinkedList;
-import java.util.List;
+import java.util.*;
 
 public class ProductBasket {
-    private List<Product> products;
+    private final Map<String, ArrayList<Product>> products;
```

Comment:
не забываем про полиморфизм
```suggestion
    private final Map<String, List<Product>> products;
```